Backport the sysdeps/x86/isa-level.h changes from:

commit ceabdcd130ca7043b0fcf2676183d79431d10493
Author: Noah Goldstein <goldstein.w.n@gmail.com>
Date:   Wed Jul 13 16:32:59 2022 -0700

    x86: Add support to build strcmp/strlen/strchr with explicit ISA level
    
    1. Add default ISA level selection in non-multiarch/rtld
       implementations.
    
    2. Add ISA level build guards to different implementations.
        - I.e strcmp-avx2.S which is ISA level 3 will only build if
          compiled ISA level <= 3. Otherwise there is no reason to
          include it as we will always use one of the ISA level 4
          implementations (strcmp-evex.S).
    
    3. Refactor the ifunc selector and ifunc implementation list to use
       the ISA level aware wrapper macros that allow functions below the
       compiled ISA level (with a guranteed replacement) to be skipped.
    
    Tested with and without multiarch on x86_64 for ISA levels:
    {generic, x86-64-v2, x86-64-v3, x86-64-v4}
    
    And m32 with and without multiarch.

Conflicts:
	support/xpthread_cond_signal.c
	sysdeps/x86_64/memrchr.S
	sysdeps/x86_64/multiarch/Makefile
	sysdeps/x86_64/multiarch/ifunc-avx2.h
	sysdeps/x86_64/multiarch/ifunc-impl-list.c
	sysdeps/x86_64/multiarch/ifunc-strcasecmp.h
	sysdeps/x86_64/multiarch/ifunc-wcslen.h
	sysdeps/x86_64/multiarch/memrchr-sse2.S
	sysdeps/x86_64/multiarch/strcasecmp_l-sse2.S
	sysdeps/x86_64/multiarch/strchr-sse2.S
	sysdeps/x86_64/multiarch/strchrnul-sse2.S
	sysdeps/x86_64/multiarch/strcmp-avx2.S
	sysdeps/x86_64/multiarch/strcmp-evex.S
	sysdeps/x86_64/multiarch/strcmp-sse2.S
	sysdeps/x86_64/multiarch/strcmp-sse4_2.S
	sysdeps/x86_64/multiarch/strcmp.c
	sysdeps/x86_64/multiarch/strlen-sse2.S
	sysdeps/x86_64/multiarch/strncmp.c
	sysdeps/x86_64/multiarch/strnlen-sse2.S
	sysdeps/x86_64/multiarch/strrchr-sse2.S
	sysdeps/x86_64/multiarch/wcschr-sse2.S
	sysdeps/x86_64/multiarch/wcscmp-sse2.S
	sysdeps/x86_64/multiarch/wcslen-sse2.S
	sysdeps/x86_64/multiarch/wcslen-sse4_1.S
	sysdeps/x86_64/multiarch/wcsncmp-sse2.c
	sysdeps/x86_64/multiarch/wcsnlen-sse4_1.S
	sysdeps/x86_64/multiarch/wcsrchr-sse2.S
	sysdeps/x86_64/strcasecmp_l.S
	sysdeps/x86_64/strchr.S
	sysdeps/x86_64/strchrnul.S
	sysdeps/x86_64/strcmp.S
	sysdeps/x86_64/strlen.S
	sysdeps/x86_64/strncase_l.S
	sysdeps/x86_64/strncmp.S
	sysdeps/x86_64/strnlen.S
	sysdeps/x86_64/strrchr.S
	sysdeps/x86_64/wcschr.S
	sysdeps/x86_64/wcscmp.S
	sysdeps/x86_64/wcslen.S
	sysdeps/x86_64/wcsrchr.S
	  (no backport of string function changes)

diff --git a/sysdeps/x86/isa-level.h b/sysdeps/x86/isa-level.h
index 77f9e2c0c3ccd41d..3c4480aba70193a2 100644
--- a/sysdeps/x86/isa-level.h
+++ b/sysdeps/x86/isa-level.h
@@ -84,6 +84,7 @@
 
 /* ISA level >= 2 guaranteed includes.  */
 #define SSE4_2_X86_ISA_LEVEL 2
+#define SSE4_1_X86_ISA_LEVEL 2
 #define SSSE3_X86_ISA_LEVEL 2
 
 
@@ -101,9 +102,18 @@
    when ISA level < 3.  */
 #define Prefer_No_VZEROUPPER_X86_ISA_LEVEL 3
 
+/* NB: This feature is disable when ISA level >= 3.  All CPUs with
+   this feature don't run on glibc built with ISA level >= 3.  */
+#define Slow_SSE42_X86_ISA_LEVEL 3
+
 /* Feature(s) enabled when ISA level >= 2.  */
 #define Fast_Unaligned_Load_X86_ISA_LEVEL 2
 
+/* NB: This feature is disable when ISA level >= 2, which was enabled
+   for the early Atom CPUs.  */
+#define Slow_BSF_X86_ISA_LEVEL 2
+
+
 /* Both X86_ISA_CPU_FEATURE_USABLE_P and X86_ISA_CPU_FEATURES_ARCH_P
    macros are wrappers for the respective CPU_FEATURE{S}_{USABLE|ARCH}_P
    runtime checks.  They differ in two ways.
